File: C++/881.cpp
__________________________________________________________________________________________________
sample 68 ms submission
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int result=0;
        sort(people.begin(),people.end());
        int left=0,right=people.size()-1;
        while(people[right]==limit){
            result++;
            right--;
        }
        while(left<right){
            int tmp=people[left]+people[right];
            if(tmp>limit){
                result++;
                right--;
            }else{
                result++;
                right--;
                left++;
            }
        }
        if(left==right)result++;
        return result;
    }
};
static auto speedup = [](){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    return nullptr;
}();
__________________________________________________________________________________________________
sample 14016 kb submission
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) 
    {
        int boats = 0;
        sort(people.begin(), people.end());
        int size = people.size();
        for (int i = 0, j = size-1; i <= j; --j)
        {
            if ((people[i] + people[j]) <= limit)
            {
                ++i;
            }
            ++boats;
        }
        
        return boats;
    }
};
__________________________________________________________________________________________________

File: Python3/881.py
__________________________________________________________________________________________________
sample 536 ms submission
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people = sorted(people)
        ans, start, end = 0, 0, len(people)-1
        while start < end:
            if people[end] + people[start] <= limit:
                start += 1
            ans += 1
            end -= 1
        return ans+1 if start == end else ans
__________________________________________________________________________________________________
sample 17560 kb submission
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        
        count = 0
        p, q = 0, len(people) - 1
        while p <= q:
            if people[p] + people[q] <= limit:
                p += 1
                q -= 1
            else:
                q -= 1
            count += 1
        return count
__________________________________________________________________________________________________
